src/dune_rules/install_entry.ml

```diff
@@ -26,36 +61,55 @@ module File = struct
         File_binding file_binding
       in
       let glob_files_decode =
-        let version_check = Dune_lang.Syntax.since Stanza.syntax (3, 6) in
-        let+ glob_files =
-          sum
-            [ ( "glob_files"
-              , let+ glob = version_check >>> String_with_vars.decode in
-                { Dep_conf.Glob_files.glob; recursive = false } )
-            ; ( "glob_files_rec"
-              , let+ glob = version_check >>> String_with_vars.decode in
-                { Dep_conf.Glob_files.glob; recursive = true } )
-            ]
-        in
+        let+ glob_files = Glob_files_with_optional_prefix.decode in
         Glob_files glob_files
       in
       file_binding_decode <|> glob_files_decode
     ;;
 
-    let to_file_bindings_unexpanded t ~expand_str ~dir =
+    let to_file_bindings_unexpanded t ~expand_str ~dir ~dune_syntax =
       match t with
       | File_binding file_binding -> Memo.return [ file_binding ]
-      | Glob_files glob_files ->
+      | Glob_files { glob_files; prefix } ->
         let open Memo.O in
-        let+ paths = Glob_files_expand.memo glob_files ~f:expand_str ~base_dir:dir in
+        let* glob_expanded =
+          Glob_files_expand.memo glob_files ~f:expand_str ~base_dir:dir
+        in
         let glob_loc = String_with_vars.loc glob_files.glob in
-        List.map paths ~f:(fun path ->
+        let glob_prefix = Glob_files_expand.Expanded.prefix glob_expanded in
+        let+ prefix_loc_opt =
+          Memo.Option.map prefix ~f:(fun prefix_sw ->
+            let+ prefix = expand_str prefix_sw in
+            prefix, String_with_vars.loc prefix_sw)
+        in
+        List.map (Glob_files_expand.Expanded.matches glob_expanded) ~f:(fun path ->
           let src = glob_loc, path in
-          File_binding.Unexpanded.make ~src ~dst:src)
+          let dst =
+            match prefix_loc_opt with
+            | None -> src
+            | Some (prefix, prefix_loc) ->
+              let path_without_prefix =
+                String.drop_prefix path ~prefix:glob_prefix
+                |> function
```

Comment:
This is the same as writing:

```
match String.drop_prefix path ~prefix:glob_prefix with
```

I'm not against using `|>`, but there's no need to favor it over basic constructs like `match`.